build.gradle

```
@@ -13,6 +13,8 @@ dependencies {
     testImplementation platform('org.assertj:assertj-bom:3.25.1')
     testImplementation('org.junit.jupiter:junit-jupiter')
     testImplementation('org.assertj:assertj-core')
+    implementation("com.mysql:mysql-connector-j:8.3.0")
+
```

```suggestion
    implementation("com.mysql:mysql-connector-j:8.3.0")
```
implementation으로 올리신 이유가 있으신가요~?
api, runtimeOnly, compileOnly에 대해 학습해봅시다 😁

---

우와 3개의 차이에 대해 전혀 신경 쓰지 않고 있었는데 이번 기회로 생각지 못했던 지식을 얻을 수 있어요!

대략적으로 알아본 결과
+ `api`는 컴파일 시 모듈이 종속적인 것까지 함께 컴파일 해줍니다.
+ `implementation`은 모듈이 의존하는 다른 모듈은 컴파일은 하지 않습니다. 즉 간접적으로 다른 모듈들의 의존성을 숨길 수 있습니다
+ `runtimeOnly`는 모듈이 runtime 에만 사용한다는 의미로 불필요한 컴파일을 방지 할 수 있습니다.
+ `CompileOnly`는 runtime과 반대로 Compile 시간에만 의존성을 사용한다는 의미입니다.

혹시 잘못된 부분이 있다거나 잘못된것이 있다면 말씀해주시면 감사하겠습니다 ☺️